### ISSUE TITLE
Data tables doesn't honor $primaryKey?

### ISSUE THREAD

**Issue by HeartlandTechie** [yajra/laravel-datatables#339]
I'm specifying a $primaryKey in my model but I'm thinking this code is ignoring that:
 
public function ordering()
    {
        foreach ($this->request->orderableColumns() as $orderable) {
            $column = $this->setupOrderColumn($orderable);
            if (isset($this->columnDef['order'][$column])) {
                $method     = $this->columnDef['order'][$column]['method'];
                $parameters = $this->columnDef['order'][$column]['parameters'];
                $this->compileColumnQuery(
                    $this->getQueryBuilder(), $method, $parameters, $column, $orderable['direction']
                );
            } else {
                /**
                 * If we perform a select("*"), the ORDER BY clause will look like this:
                 * ORDER BY * ASC
                 * which causes a query exception
                 * The temporary fix is modify `*` column to `id` column
                 */
                if ($column === '*') {
                    $column = 'id';
                }
                $this->getQueryBuilder()->orderBy($column, $orderable['direction']);
            }
        }
    }

**Comment by yajra** [comment]
Can you please paste some snippets on how you declare your dataTable? As stated in the comment, it's a temporary fix that uses id as the primary key. If you have a better solution in mind, you can also submit a PR. Thanks!

**Comment by yajra** [comment]
Just added a possible fix. Can you try using dev-develop branch and confirm if issue was fixed? Thanks

**Comment by HeartlandTechie** [comment]
Thanks will try it today

**Comment by HeartlandTechie** [comment]
Tried it, didn't exactly help, but I modified how I was selecting the data and it seems to have helped.  Will continue researching on my end a bit more.

**Comment by yajra** [comment]
Are you still having this issue? Thanks!

**Comment by ChaosPower** [comment]
@yajra and @HeartlandTechie got the error as well. This issue almost looks like it is not and it is an issue for Laravel Datatables (sort of like a false positive thing). 

datatables (js) itself is querying dynamically base on the column data / name. Laravel Datatables only fetches or gets the query variables or post vars from the client especially when it comes to ordering. I was also wondering why all columns are ordering by ID. But then again, I think something or somewhere is translating all with suffix ID (yourID) as ID. 

That something is removing is disregarding other characters when it has detected ID on the column data.

##### Laravel Datatables does not ignore primary keys. The reason why it ignored it is because it was sent or determined by the client (js).

Base on the source, ordering function's columns are dependent on the data passed to it **(query vars / post vars)**.

However, this can be avoided by defining your columns at the client side.

```
    var dt = $('.dt').DataTable({
        serverSide: true,
        ajax: '{!! route('advances.index') !!}',
        columns: [
            {data: 'agencyid', name: 'agencyid', title: 'Agency ID'},
            {data: 'advanceid', name: 'advanceid', title: 'Advance ID'}
        ]
    });

```

Below is my snippet to replicate this issue.

Model
```
use Illuminate\Database\Eloquent\Model;

class Advance extends Model
{
    protected $table = 'advance';
    protected $primaryKey = 'advanceid';
    public static $snakeAttributes = false;
    const CREATED_AT = 'created_at';
    const UPDATED_AT = 'updated_at';
}
```
Controller
```
    public function index(Request $request)
    {
        $advances = new Advance();

        if ($request->ajax()) {
            return $this->datatables->of($advances->select(['*']))->make(true);
        }

        return view('advances.index');
    }

```

HTML
```
<script>
    var dt = $('.dt').DataTable({
        serverSide: true,
        ajax: '{!! route('advances.index') !!}',
        columns: [
            {data: 'agencyid', name: 'agencyid', title: 'Agency ID'},
            {data: 'advanceid', name: 'advanceid', title: 'Advance ID'}
        ]
    });
</script>
```

**Comment by yajra** [comment]
Guys, can you test dev-master? I think I found what was causing the issue. Please confirm if fixed. Thanks!

**Comment by ChaosPower** [comment]
@yajra the new feature is working. Thanks for adding it.

**Comment by yajra** [comment]
Thanks for confirming @ChaosPower. Fix released on v6.5.1
